test/e2e/node/kubelet.go

```diff
@@ -479,62 +485,266 @@ var _ = SIGDescribe("kubelet", func() {
 
 		/*
 			Test if kubectl node-logs <node-name>
-			returns something or not!
+			returns an error or not!
 		*/
 
-		ginkgo.It("should return the logs ", func() {
+		ginkgo.It("should return the error of running without --query option", func() {
 			ginkgo.By("Starting the command")
 			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
 
 			for nodeName := range nodeNames {
 				cmd := tk.KubectlCmd("alpha", "node-logs", nodeName)
-				runKubectlCommand(cmd, "")
+				_, _, err := framework.StartCmdAndStreamOutput(cmd)
+				if err != nil {
+					framework.Failf("Failed to start kubectl command: %v", err)
+				}
+				err = cmd.Wait()
+				if err == nil {
+					framework.Failf("Command failed: %v. It was expected an error!", cmd)
+				}
 			}
 		})
 
 		/*
-			Test if kubectl node-logs <node-name> --service kubelet
-			returns something or not!
+			Test if kubectl node-logs <node-name> --query kubelet
+			returns the kubelet logs or not!
 		*/
 
-		ginkgo.It("should return the logs for the requested service", func() {
+		ginkgo.It("should return the logs for the requested query", func() {
 			ginkgo.By("Starting the command")
 			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
 
 			for nodeName := range nodeNames {
-				cmd := tk.KubectlCmd("alpha", "node-logs", nodeName, "--service", "kubelet")
-				runKubectlCommand(cmd, "--service")
+				cmd := tk.KubectlCmd("alpha", "node-logs", nodeName, "--query", "kubelet")
+				result := runKubectlCommand(cmd, "--query")
+				lookForString("kubelet", result)
 			}
 		})
 
 		/*
-			Test if kubectl node-logs <node-name> --path pods
-			returns something or not!
+			Test if kubectl node-logs --role master --query kubelet --tail 4 -o short
+			returns 4 lines from the kubelet logs or not!
 		*/
 
-		ginkgo.It("should return the logs for the provided path", func() {
+		ginkgo.It("should return the first 4 lines of logs for the requested query of the specific role", func() {
 			ginkgo.By("Starting the command")
+			n := "4"
 			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "--tail", n, "-o", "short")
+			result := runKubectlCommand(cmd, "--role --query --tail")
+			logs := journalctlCommand("-u", "kubelet", "-n", n, "")
+			if result != logs {
+				framework.Failf("Failed to receive the correct kubelet logs or the correct amount of lines of logs")
```

Comment:
would be useful to output both compared values to simplify test troubleshooting

Reply:
`runKubectlCommand` already logs the output. I will modify the message and add a framework.Logf to the journalctl function as well